oak-core/src/main/java/org/apache/jackrabbit/oak/query/xpath/Statement.java

```diff
@@ -376,6 +376,21 @@ private static void appendQueryOptions(StringBuilder buff, QueryOptions queryOpt
             buff.append("]");
             optionCount++;
         }
+        if (queryOptions.offset != -1) {
+            if (optionCount > 0) {
+                buff.append(", ");
+            }
+            buff.append("offset ");
+            buff.append(queryOptions.offset);
+            optionCount++;
+        }
+        if (queryOptions.limit != -1) {
+            if (optionCount > 0) {
+                buff.append(", ");
+            }
+            buff.append("limit ");
+            buff.append(queryOptions.limit);
```

Comment:
missing optionCount++... I know, it's not needed here really, but it's easy to forget to add it later...
Actually, it's probably better to add the options first into an ArrayList<String>, and then use String.join(", ", list)...

Reply:
Makes sense, I updated to add to a list and then join the list.